core/unit_test/TestConcepts.hpp

```diff
@@ -59,4 +59,314 @@ static_assert(std::is_same<int, Kokkos::Impl::remove_cvref_t<int &>>{}, "");
+
+  template <class U>
+  using LeagueSizeArchetypeExpr =
+      decltype(std::declval<U const &>().league_size());
+
+  // scratch space
+  template <class U>
+  using TeamShmemArchetypeExpr =
+      decltype(std::declval<U const &>().team_shmem());
+
+  template <class U>
+  using TeamScratchArchetypeExpr =
+      decltype(std::declval<U const &>().team_scratch(int{}));
+
+  template <class U>
+  using ThreadScracthArchetypeExpr =
+      decltype(std::declval<U const &>().thread_scratch(int{}));
+
+  // collectives
+  template <class U>
+  using TeamBarrierArchetypeExpr =
+      decltype(std::declval<U const &>().team_barrier());
+
+  template <class U>
+  using TeamBroadcastArchetypeExpr =
+      decltype(std::declval<U const &>().team_broadcast(
+          lvalueForMethodsNeedingIt_, int{}));
+
+  template <class U>
+  using TeamBroadcastAcceptClosureArchetypeExpr =
+      decltype(std::declval<U const &>().team_broadcast(
+          TrivialFunctor{}, lvalueForMethodsNeedingIt_, int{}));
+
+  template <class U>
+  using TeamReducedArchetypeExpr =
+      decltype(std::declval<U const &>().team_reduce(
+          std::declval<reduction_to_test_t>()));
+
+  template <class U>
+  using TeamScanArchetypeExpr = decltype(std::declval<U const &>().team_scan(
+      lvalueForMethodsNeedingIt_, ptrForMethodsNeedingIt_));
+
+ public:
+  static constexpr bool value =
+      Kokkos::is_detected_v<ExecutionSpaceArchetypeAlias, T> &&
+      Kokkos::is_detected_v<ScratchMemorySpaceArchetypeAlias, T> &&
+      //
+      Kokkos::is_detected_exact_v<int, TeamRankArchetypeExpr, T> &&
+      Kokkos::is_detected_exact_v<int, TeamSizeArchetypeExpr, T> &&
+      Kokkos::is_detected_exact_v<int, LeagueRankArchetypeExpr, T> &&
+      Kokkos::is_detected_exact_v<int, LeagueSizeArchetypeExpr, T> &&
+      //
+      Kokkos::is_detected_exact_v<
+          Kokkos::detected_t<ScratchMemorySpaceArchetypeAlias, T> const &,
+          TeamShmemArchetypeExpr, T> &&
+      Kokkos::is_detected_exact_v<
+          Kokkos::detected_t<ScratchMemorySpaceArchetypeAlias, T> const &,
+          TeamScratchArchetypeExpr, T> &&
+      Kokkos::is_detected_exact_v<
+          Kokkos::detected_t<ScratchMemorySpaceArchetypeAlias, T> const &,
+          ThreadScracthArchetypeExpr, T> &&
+      //
+      Kokkos::is_detected_exact_v<void, TeamBarrierArchetypeExpr, T> &&
+      Kokkos::is_detected_exact_v<void, TeamBroadcastArchetypeExpr, T> &&
+      Kokkos::is_detected_exact_v<void, TeamBroadcastAcceptClosureArchetypeExpr,
+                                  T> &&
+      Kokkos::is_detected_exact_v<void, TeamReducedArchetypeExpr, T> &&
+      Kokkos::is_detected_exact_v<test_value_type, TeamScanArchetypeExpr, T>;
+  constexpr operator bool() const noexcept { return value; }
+};
+
+template <class T>
+inline constexpr bool is_team_handle_complete_trait_check_v =
+    is_team_handle_complete_trait_check<T>::value;
+
+// actual test begins here
```

Comment:
I don't think we need all of this stuff here, basically you wrote a test for your test or?

Reply:
@crtrott no the full trait is there because originally that was in the implementation, but you, Damien and I discussed it and decided to leave it inside the test and not throw it away

Reply:
does it ring a bell?

Reply:
also, just FYI, the OpenMPTARGET team handle does *not* meet the team handle concept documented in [here](https://kokkos.github.io/kokkos-core-wiki/API/core/policies/TeamHandleConcept.html) and I was able to discover that only using my *full* trait thing, see my comment below for excluding the OpenMPTarget from the check

Reply:
@crtrott I clarified this with Damien, I misundertsood your comment , will fix this